File: sdks/ts/src/sdk/models/shared/keywordskillcard.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



// KeywordSkillCardEntries
/** 
 * An entry in the `entries` attribute of a metadata card
**/
export class KeywordSkillCardEntries extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=text" })
  text?: string;
}

export enum KeywordSkillCardInvocationTypeEnum {
    SkillInvocation = "skill_invocation"
}


// KeywordSkillCardInvocation
/** 
 * The invocation of this service, used to track
 * which instance of a service applied the metadata.
**/
export class KeywordSkillCardInvocation extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=id" })
  id: string;

  @SpeakeasyMetadata({ data: "json, name=type" })
  type: KeywordSkillCardInvocationTypeEnum;
}

export enum KeywordSkillCardSkillTypeEnum {
    Service = "service"
}


// KeywordSkillCardSkill
/** 
 * The service that applied this metadata.
**/
export class KeywordSkillCardSkill extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=id" })
  id: string;

  @SpeakeasyMetadata({ data: "json, name=type" })
  type: KeywordSkillCardSkillTypeEnum;
}


// KeywordSkillCardSkillCardTitle
/** 
 * The title of the card.
**/
export class KeywordSkillCardSkillCardTitle extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=code" })
  code?: string;

  @SpeakeasyMetadata({ data: "json, name=message" })
  message: string;
}

export enum KeywordSkillCardSkillCardTypeEnum {
    Keyword = "keyword"
}

export enum KeywordSkillCardTypeEnum {
    SkillCard = "skill_card"
}


// KeywordSkillCard
/** 
 * A skill card that contains a set of keywords
**/
export class KeywordSkillCard extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=created_at" })
  createdAt?: Date;

  @SpeakeasyMetadata({ data: "json, name=entries", elemType: KeywordSkillCardEntries })
  entries: KeywordSkillCardEntries[];

  @SpeakeasyMetadata({ data: "json, name=invocation" })
  invocation: KeywordSkillCardInvocation;

  @SpeakeasyMetadata({ data: "json, name=skill" })
  skill: KeywordSkillCardSkill;

  @SpeakeasyMetadata({ data: "json, name=skill_card_title" })
  skillCardTitle?: KeywordSkillCardSkillCardTitle;

  @SpeakeasyMetadata({ data: "json, name=skill_card_type" })
  skillCardType: KeywordSkillCardSkillCardTypeEnum;

  @SpeakeasyMetadata({ data: "json, name=type" })
  type: KeywordSkillCardTypeEnum;
}

File: python-sdk/src/boxsdk/models/shared/keywordskillcard.py
import dataclasses
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from boxsdk import utils


@dataclass_json
@dataclasses.dataclass
class KeywordSkillCardEntries:
    r"""KeywordSkillCardEntries
    An entry in the `entries` attribute of a metadata card
    """
    
    text: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('text') }})
    
class KeywordSkillCardInvocationTypeEnum(str, Enum):
    SKILL_INVOCATION = "skill_invocation"


@dataclass_json
@dataclasses.dataclass
class KeywordSkillCardInvocation:
    r"""KeywordSkillCardInvocation
    The invocation of this service, used to track
    which instance of a service applied the metadata.
    """
    
    id: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    type: KeywordSkillCardInvocationTypeEnum = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('type') }})
    
class KeywordSkillCardSkillTypeEnum(str, Enum):
    SERVICE = "service"


@dataclass_json
@dataclasses.dataclass
class KeywordSkillCardSkill:
    r"""KeywordSkillCardSkill
    The service that applied this metadata.
    """
    
    id: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    type: KeywordSkillCardSkillTypeEnum = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('type') }})
    

@dataclass_json
@dataclasses.dataclass
class KeywordSkillCardSkillCardTitle:
    r"""KeywordSkillCardSkillCardTitle
    The title of the card.
    """
    
    message: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('message') }})
    code: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('code') }})
    
class KeywordSkillCardSkillCardTypeEnum(str, Enum):
    KEYWORD = "keyword"

class KeywordSkillCardTypeEnum(str, Enum):
    SKILL_CARD = "skill_card"


@dataclass_json
@dataclasses.dataclass
class KeywordSkillCard:
    r"""KeywordSkillCard
    A skill card that contains a set of keywords
    """
    
    entries: list[KeywordSkillCardEntries] = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('entries') }})
    invocation: KeywordSkillCardInvocation = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('invocation') }})
    skill: KeywordSkillCardSkill = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('skill') }})
    skill_card_type: KeywordSkillCardSkillCardTypeEnum = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('skill_card_type') }})
    type: KeywordSkillCardTypeEnum = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('type') }})
    created_at: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('created_at'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    skill_card_title: Optional[KeywordSkillCardSkillCardTitle] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('skill_card_title') }})
    
